File: Python/Problems/Problems/problem1550.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem


class Problem1550(Problem):
    def test(self) -> bool:
        arr = [2, 6, 4, 1]

        expected = False

        result = self.threeConsecutiveOdds(arr)

        return expected == result

    def threeConsecutiveOdds(self, arr: list[int]) -> bool:
        n = len(arr)
        for i in range(2, n):
            if (arr[i - 2] & 1) == 1 and (arr[i - 1] & 1) == 1 and (arr[i] & 1) == 1:
                return True

        return False

File: C++/Problems/Problems/problem_1550.cpp
#include "problem_1550.h"

bool problem_1550::test() {
    std::vector arr  { 2, 6, 4, 1 };

    bool expected = false;

    auto result = threeConsecutiveOdds(arr);

    return expected == result;
}

bool problem_1550::threeConsecutiveOdds(std::vector<int> &arr) {
    int n = arr.size();
    for (int i = 2; i < n; ++i)
    {
        if ((arr[i - 2] & 1) == 1 && (arr[i - 1] & 1) == 1 && (arr[i] & 1) == 1)
            return true;
    }

    return false;
}
